### DISCUSSION TITLE
Replacement for Mapsforge.map.layer.overlay.Circle?

### DISCUSSION THREAD

**Discussion by moving-bits** [mapsforge/vtm#914] [Q&A] [answered]
Making some progress along the way, putting the first items on my maps now...

In `mapsforge/mapsforge` there is a nice overlay `Circle` which you could give a latLong and a radius to draw a circle (optionally filled). Is there some sort of replacement for this (couldn't find one) or do you have a pointer for me where to start? 

For drawing a heading indicator I use an ItemizedLayer and add a marker with the rotated arrow. For updating this marker (on heading change) I need to remove the item and add a new marker, just calling `setMarker()` on an already existing item did not work for me. Is this the right approach, or have I missed something?

Thanks.

**Comment by devemux86** [comment, marked as answer]
> Circle

You can use the [PointDrawable](https://github.com/mapsforge/vtm/blob/master/vtm-jts/src/org/oscim/layers/vector/geometries/PointDrawable.java) with buffer or the [CircleDrawable](https://github.com/mapsforge/vtm/blob/master/vtm-jts/src/org/oscim/layers/vector/geometries/CircleDrawable.java).
See the [VectorLayerActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/VectorLayerActivity.java) sample for the 1st option and how to use the various JTS drawables.

See also the https://github.com/mapsforge/vtm/discussions/861 discussion.

---

> heading indicator

For location marker / arrow see:
- [LocationTextureActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/LocationTextureActivity.java) sample using bitmaps
- [LocationActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/LocationActivity.java) sample using OpenGL shaders

**Reply by moving-bits** [reply]
Thanks for the pointers. I had excluded some of the vtm subfolders (including JTS) during my search, thinking it was some very specific adapter. :-(
But Circle is working now. Will look into LocationTextureActivity next to see if using that is smoother than my current solution for location/heading indicator.

**Comment by devemux86** [comment]
> I had excluded some of the vtm subfolders (including JTS) during my search, thinking it was some very specific adapter.

It is needed for all types of overlays in VTM using [JTS geometries](https://github.com/mapsforge/vtm/blob/master/docs/Integration.md#jts-geometries).
